packages/amplify-graphql-model-transformer/src/resources/custom-resource-lambda/custom-resource-handler.ts

```diff
@@ -0,0 +1,440 @@
+ * Such as 'true' to true, '5' to 5
+ * @param obj Input object
+ * @returns Oject with its values converted to the correct form of boolean or number
+ */
+const convertStringToBooleanOrNumber = (obj: Record<string, any>): Record<string, any> => {
+  for (const key in obj) {
+    if (typeof obj[key] === 'object') {
+      // If the property is an object, recursively call the function
+      obj[key] = convertStringToBooleanOrNumber(obj[key]);
+    } else if (typeof obj[key] === 'string') {
+      if (obj[key] === 'true' || obj[key] === 'false') {
+        // If the property is a string with value 'true' or 'false', convert it to a boolean
+        obj[key] = obj[key] === 'true';
+      } else if (!isNaN(Number(obj[key]))) {
+        // If the property is a string that can be parsed into a number, convert it to a number
+        obj[key] = Number(obj[key]);
+      }
+    }
+  }
+  return obj;
+};
+/**
+ * Util function to remove undefined values from root level of object
+ * @param obj Input Object
+ * @returns obj without undefined attributes
+ */
+const removeUndefinedAttributes = (obj: Record<string, any>): Record<string, any> => {
+  for (const key in obj) {
+    if (obj[key] === undefined) {
+      // Use the delete operator to remove the attribute if it's undefined
+      delete obj[key];
+    }
+  }
+  return obj;
+};
+
+const toCreateTableInput = (props: any): CreateTableInput => {
+  const createTableInput: CreateTableInput = {
+    TableName: props.TableName,
+    AttributeDefinitions: props.AttributeDefinitions,
+    KeySchema: props.KeySchema,
+    GlobalSecondaryIndexes: props.GlobalSecondaryIndexes,
+    BillingMode: props.BillingMode,
+    StreamSpecification: props.StreamSpecification
+      ? {
+          StreamEnabled: true,
+          StreamViewType: props.StreamSpecification.StreamViewType,
+        }
+      : undefined,
+    ProvisionedThroughput: props.ProvisionedThroughput,
+    SSESpecification: props.SSESpecification ? { Enabled: props.SSESpecification.sseEnabled } : undefined,
+  };
+  return removeUndefinedAttributes(createTableInput) as CreateTableInput;
+};
+
+const createNewTable = async (input: CreateTableInput): Promise<CreateTableResponse> => {
+  const tableName = input.TableName;
+  const createTableInput: CreateTableInput = input;
+  const result = await ddbClient.createTable(createTableInput).promise();
+  return { tableName, tableArn: result.TableDescription?.TableArn!, streamArn: result.TableDescription?.LatestStreamArn };
+};
+
+const doesTableExist = async (tableName: string): Promise<boolean> => {
+  try {
+    await ddbClient.describeTable({ TableName: tableName }).promise();
+    return true; // Table exists
+  } catch (error) {
+    if (error.code === 'ResourceNotFoundException') {
+      return false; // Table does not exist
+    }
+    throw error; // Handle other errors
+  }
+};
+
+const isProjectionModified = (currentProjection: Projection, endProjection: Projection): boolean => {
```

Comment:
ANy reason this isn't simply an "equals" function?

Reply:
What is the difference here by the `equals` function you mentioned?

Reply:
No functional difference, it's just a naming thing: traditionally a function that checks for equality would be named to reflect that. Not a big deal now, but if you end up using it elsewhere I'd suggest renaming (and changing the condition check) to make its usage and form more conventional. Feel free to resolve this conversation.